tools/lint-pr-url.mjs

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env node
+
+// Usage:
+// git diff upstream/master HEAD -G"pr-url:" -- "*.md" | \
+// ./tools/lint-pr-url.mjs <expected-pr-url>
+
+import process from 'node:process';
+import readline from 'node:readline';
+
+const [, , expectedPrUrl] = process.argv;
+
+const fileDelimiter = /^\+\+\+ b\/(.+\.md)$/;
+const changeDelimiter = /^@@ -\d+,\d+ \+(\d+),\d+ @@/;
+const prUrlDefinition = /^\+\s+pr-url: (.+)$/;
+
+let currentFile;
+let currentLine;
+
+const validatePrUrl = (url) => url == null || url === expectedPrUrl;
+
+const diff = readline.createInterface({ input: process.stdin });
+for await (const line of diff) {
+  if (fileDelimiter.test(line)) {
+    currentFile = line.match(fileDelimiter)[1];
+    console.log(`Parsing changes in ${currentFile}.`);
+  } else if (changeDelimiter.test(line)) {
+    currentLine = Number(line.match(changeDelimiter)[1]) - 1;
+  } else if (!validatePrUrl(line.match(prUrlDefinition)?.[1])) {
+    console.warn(
+      `::warning file=${currentFile},line=${currentLine++},col=${line.length}` +
```

Comment:
```suggestion
      `::warning file=${currentFile},line=${++currentLine},col=${line.length}` +
```
Otherwise it refers to the previous line.

Reply:
Good catch! I'm going to remove the `- 1` in the line above instead.